instance/gandi.py

```diff
@@ -189,4 +192,157 @@ def remove_dns_record_callback(zone_id, zone_version):
         )
 
 
-api = GandiAPI()
+class GandiV5API:
+    """
+    Gandi v5 LiveDNS API client
+    """
+    def __init__(self, api_key, api_base='https://dns.api.gandi.net/api/v5'):
+        self.api_base = api_base
+        self.api_key = api_key
+        self._domain_cache = None
+
+    def _populate_domain_cache(self):
+        """
+        Populate the internal domain cache with all domains in the current Gandi account.
+        """
+        self._domain_cache = []
+        domains_api_url = '{}/domains'.format(self.api_base)
+
+        try:
```

Comment:
Do you think it's worth trying to upstream this method of retrieving domains in account to Lexicon? If so, and they already have some sort of support for it on other providers, can you create an upstream ticket?

Reply:
@giovannicimolin, the domain cache is needed only to split the given hostname into the registered domain name and its subdomain in the `_split_domain_name()` method and that's why it is called there before performing the split operation.

Reply:
> Do you think it's worth trying to upstream this method of retrieving domains in account to Lexicon? If so, and they already have some sort of support for it on other providers,

@giovannicimolin, `lexicon` explicitly mentions that it is for manipulating DNS records. From what I can see, all the lexicon operations require a domain name. I checked the [base argparse configuration](https://github.com/AnalogJ/lexicon/blob/master/lexicon/parser.py#L9) for the CLI and also some of the main providers and couldn't find this sort of operation supported anywhere. 

Hence imho this feels out of scope for upstreaming, It is just a simple `GET` API call and can be done in our code itself.

Reply:
Makes sense.